webapp/src/components/Asset/Asset.js

```diff
@@ -83,6 +80,45 @@ export default class Asset extends React.PureComponent {
 
   componentWillUnmount() {
     isNavigatingAway = false
+    isAssetPrefetched = false
+  }
+
+  isValid(wallet, asset) {
+    return (
+      this.isValidOwnership(wallet, asset) &&
+      this.isValidRole(wallet, asset) &&
+      this.isValidAssetState(asset)
+    )
+  }
+
+  isValidOwnership(wallet, asset) {
+    const { shouldDisallowOwner, shouldBeOwner } = this.props
+    const isAssetOwner = isOwner(wallet.address, asset)
+
+    let ownership = true
+
+    if (shouldBeOwner !== undefined) {
+      ownership = ownership && (isAssetOwner && shouldBeOwner)
```

Comment:
**Note**: I renamed `ownership` with isValid to keep consistent with boolean flags.

Hmm not really needed, I added it there to be consistent. I think this:

```ts
let isValid = true

if (shouldBeOwner !== undefined) {
  isValid = isValid && (isAssetOwner && shouldBeOwner)
}

if (shouldDisallowOwner !== undefined) {
  isValid = isValid && (!isAssetOwner && !!asset && shouldDisallowOwner)
}

return isValid
```

might be a bit more straightforward

than reading:

 ```ts
let isValid = true

if (shouldBeOwner !== undefined) {
  isValid = isAssetOwner && shouldBeOwner
}

if (shouldDisallowOwner !== undefined) {
  isValid = isValid && (!isAssetOwner && !!asset && shouldDisallowOwner)
}

return isValid
```

Also it prevents bugs if someone adds something above `if (shouldBeOwner !== undefined)` or if that condition is moved.

What do you think? Not trying to convince you here

Reply:
Can the flow enter into both conditions at the same time?

`(shouldBeOwner !== undefined)` and `(shouldDisallowOwner !== undefined)`

Maybe we should check that both properties can not be true at the same time and that is why I said that `isValid` will always be true.

Reply:
It can, but only if you define both properties on the Component like this:

```ts
<Parcel shouldBeOwner shouldDisallowOwner />
```

which doesn't make sense.

I would argue that the best thing we can do here is delete `shouldDisallowOwner`. I'm not sure why it was there in the first place.
Am I wrong to say that it's the same as: `shouldBeOwner={false}` ?

Reply:
Hmmm, not at 100%. `shouldBeOwner={false}` could be that everyone can access including not owners (not mandatory). I would leave as it is or just throw if both conditions have been set.